GETTING_STARTED.md

```diff
@@ -0,0 +1,63 @@
+# Getting Started as a Boutiques Developer
+
+This guide will help you set up your local environment for coding and testing Boutiques.
+
+## 0. Things you'll need
+- **A Linux system**. It _may_ be possible to develop Boutiques on Windows, but will likely cause many headaches.
+- **Root (sudo) access**. Unless your system already has everything you need installed, you'll need root permission to install new programs.
+- **Python**. Boutiques supports any version >= 2.7. It's a good idea to have multiple versions of Python installed (e.g. 2.7 and 3.7) so you can test your code for version compatibility.
+- **pip**. Most versions of Python will come with it, but in case you don't have it, instructions to install it are [here](https://pip.pypa.io/en/stable/installing/).
+- **Docker**. If you don't already have Docker, follow the [instructions](https://docs.docker.com/install/overview/) to install Docker CE.
+- **Singularity**. Follow the instructions [here](https://singularity.lbl.gov/install-linux) to download the latest stable release of Singularity.
+- **Git**. If you don't have Git, instructions to install it are [here](https://git-scm.com/download/linux).
+
+## 1. Clone the boutiques repository
+- Fork the [Boutiques repository](https://github.com/boutiques/boutiques).
+- Clone your fork locally: 
+    `git clone https://github.com/<YOUR-GITHUB-USERNAME>/boutiques.git`
+
+## 2. Create a virtual environment
+- Install virtualenv:
+  `sudo pip install virtualenv`
+- Create a virtual envrionment in the Python tools root directory:
+  `cd boutiques/tools/python`
+  `virtualenv env`
+- Use the environment whenever you work on Boutiques by running:
+  `source env/bin/activate`
+- Note: if you wish to leave the virtual environment, simply run `deactivate`.
+
+## 3. Install Boutiques, test framework and validation tools
+From within your virtual environment in the `boutiques/tools/python` directory:
+- Upgrade setuptools: `pip install --upgrade setuptools`
+- Install pytest and pycodestyle: `pip install pytest pycodestyle`
+  - pytest is a unit testing framework for Python.
+  - pycodestyle checks if your code conforms to Python style conventions.
+- Install jsonlint: `npm install jsonlint -g` 
+  - In case you don't have Node installed, instructions to install it are [here](https://www.npmjs.com/get-npm).
+  - jsonlint is a tool to validate JSON files. 
+- Install Boutiques: `pip install boutiques` or just `pip install .`

```

Comment:
I'd suggest we recommend `pip install -e .` for installing in a dev environment. Advantage over `pip install boutiques` is obvious (you get the dev version), and over `.` without the `-e` is that the `-e` switch enables "editable mode", which automatically updates the library whenever you make changes to files.

Reply:
I totally didn't realize that `pip install boutiques` will just install the latest release, oops

Reply:
gold